source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+    // create a new vector 
+    let mut vec = Vec::new();
+    vec.push(Parameter::new("message", CLType::String));
+
+    // create an Entry Point Object
+    let mut entry_points = EntryPoints::new();
+
+    // add the entry point to the entry points object
+    entry_points.add_entry_point(EntryPoint::new(
+        "update_msg",                   // the name of the entry point
+        vec,                            // the arguments which can be passed into the entry point
+        CLType::Unit,                   // return type of the entry point
+        EntryPointAccess::Public,       // acces permissions - public can be accessed always
+        EntryPointType::Contract        // in most cases it will be contract
+    ));
+
+    // the contract is stored in the global state
+    let (stored_contract_hash, _contract_version) = storage::new_contract(
+        entry_points,                                       // entry points
+        Some(named_keys),                                   // named keys 
+        Some("Hello_world_package_name".to_string()),       // package name
+        Some("Hello_world_access_uref".to_string())         // access uref
+    );
+
+    // to access from the account - named keys of the account
+    runtime::put_key("hello_world_contract", stored_contract_hash.into());
+
+}
+```
+
+:::tip
+
+We should notice that that runtime and storage appear quite frequently in our code. If you didn't have a chance to check out those modules yet it would be a perfect time to do so: [contract API](https://docs.rs/casper-contract/latest/casper_contract/contract_api/index.html).
+The amount of native functions isn't that large so it might be advisable to think about which one of those could be used with your scenario.
+
+:::
+
+You should always keep the entry points and the functions of the contract in sync. If there is an entry point with name `update_msg` there should be also a function with the same name in the contract.
+
+This is presented below:
+
+```bash
+
+#[no_mangle]
+pub extern "C" fn update_msg() {
+
+    // get the argument from the runtime so from the outside call
+    let value: String = runtime::get_named_arg("message");
+
+    // get the uref of the message stored in the global state
+    let uref = runtime::get_key("message").unwrap_or_revert().into_uref().unwrap_or_revert();
+
+    // write the new message to the global state
+    storage::write(uref, String::from(value));
+}
+
+```
+
+You can always try out storing different types in the context. A distinction between storing data in the context of a smart contract and using dictionaries has to be mentioned. The dictionaries can be used to store the data which is dapp centric. Just be sure to remember that the dictionary is not an SQL Database and only what is necessary and should be stored on the blockchain can go into the dictionary. The types stored in the contracts context could be regarded more as key links between parts of a bigger application.
```

Comment:
An off-topic tip, a distraction from the task which is getting this contract up and running. We need a "tip" popout which will allow us to do this kind o thing without interrupting the flow of the tutorial.

Reply:
Introduced this part as the Info block. it is different and can be omitted.